src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -192,7 +193,31 @@ ThreadError DatasetManager::Register(void)
 
     VerifyOrExit((message = mSocket.NewMessage(0)) != NULL, error = kThreadError_NoBufs);
     SuccessOrExit(error = message->Append(header.GetBytes(), header.GetLength()));
-    SuccessOrExit(error = message->Append(mLocal.GetBytes(), mLocal.GetSize()));
+
+#if OPENTHREAD_ENABLE_COMMISSIONER
+    isCommissioner = mNetif.GetCommissioner().GetState() > MeshCoP::Commissioner::kStateDisabled ? true : false;
```

Comment:
`mNetif.GetCommissioner().GetState() != MeshCoP::Commissioner::kStateDisabled` is better?